play-services-core/src/main/kotlin/org/microg/gms/auth/signin/AuthSignInService.kt

```diff
@@ -77,10 +79,14 @@ class AuthSignInServiceImpl(
         }
         lifecycleScope.launchWhenStarted {
             try {
-                val account = account ?: options?.account ?: SignInConfigurationService.getDefaultAccount(context, packageName)
-                if (account != null && options?.isForceCodeForRefreshToken != true && options?.includeUnacceptableScope != true) {
+                val account = account 
+                    ?: options?.account 
+                    ?: SignInConfigurationService.getDefaultAccount(context, packageName) 
+                    ?: AccountManager.get(context).getAccountsByType(AuthConstants.DEFAULT_ACCOUNT_TYPE).firstOrNull()
+                Log.d(TAG, "silentSignIn: account -> ${account?.name}")
+                if (account != null && options?.isForceCodeForRefreshToken != true) {
                     if (getOAuthManager(context, packageName, options, account).isPermitted || AuthPrefs.isTrustGooglePermitted(context)) {
-                        val googleSignInAccount = performSignIn(context, packageName, options, account)
+                        val googleSignInAccount = performSignIn(context, packageName, options, account, true)
```

Comment:
Silent sign-in shouldn't be considered permitted by default (permitted means that the user actively agreed to give the app access to their account)

Reply:
Modified. When there is no permission, a pop-up window will be displayed to prompt the authorization content. It depends on the background pop-up window permission.